### DISCUSSION TITLE
Understanding the use of "~" with the "--height" option.

### DISCUSSION THREAD

**Discussion by habahabahaba** [junegunn/fzf#3989] [Q&A] [answered]
Hello. I want to understand what is the expected behavior of:
`--height=~80%`
Does this mean: the height of the **fzf** window will be such, as to display all the input, _unless_ it would take more than 80% of the screen, in which case the height of the window will be 80% ?

If my assumption is correct, could you please check if in the **fzf 0.55.0** it works for you that way?

Thank you.

**Comment by LangLangBart** [comment]
> I want to understand what is the expected behavior of

`man fzf` 
```txt
Layout
    --height=[~]HEIGHT[%]
          Display fzf window  below  the  cursor  with  the  given  height
          instead of using the full screen.

          If  a  negative  value is specified, the height is calculated as
          the terminal height minus the given value.

            fzf --height=-1

          When prefixed with  ~,  fzf  will  automatically  determine  the
          height in the range according to the input size.

            # Will not take up 100% of the screen
            seq 5 | fzf --height=~100%

          Adaptive height has the following limitations:
          *  Cannot  be  used  with top/bottom margin and padding given in
          percent size
          * Negative value is not allowed
          * It will not find the right  size  when  there  are  multi-line
          items
```

> If my assumption is correct, could you please check if in the fzf 0.55.0 it works for you that way?

It seems to work as expected. If the input requires less than 80% of the terminal, the fzf window only uses that space, but it won't exceed 80% if more space is needed. 

<img src="https://github.com/user-attachments/assets/26924073-7c8a-43f3-aec2-b96bca7dac3e" width="800">

<img src="https://github.com/user-attachments/assets/25702516-3b2b-4da0-a97c-46cf8434b428" width="800">

**Reply by habahabahaba** [reply]
Thank you, [LangLangBart](https://github.com/LangLangBart),
Can you please try your last query (`seq 40 | ...`) with this option: `--preview-window='hidden:bottom:-3`.
On my machine it makes **fzf** window to cover the whole screen. IOW, if I use `--preview=top` or `--preview=bottom`, it breaks (overrides?) the `--height=~80%` option.

**Reply by junegunn** [reply, marked as answer]
Looks like `~SIZE` only limits the size of the main window. The height of the preview window is calculated after the height of the main window is determined. I don't quite remember if I did it on purpose or not, but anyway, it is how it is. You can work around the problem by using the half of the desired height.

```sh
# The main window will take at most 40% of the screen height. The preview window will take the same height.
fzf --height=~40% --preview 'seq 1000' --preview-window top

seq 10 | fzf --height=~40% --preview 'seq 1000' --preview-window top
```

**Reply by habahabahaba** [reply]
[junegunn](https://github.com/junegunn), thank you for the advice.
I just checked and on my machine `--height=~` rules have no effect if the terminal is shorter than **20 rows**, in that case **fzf** window spans the whole 100%.
Should I file the bug report?
